### PR TITLE
Issue/27

### PR DESCRIPTION
**Pull request by adatzer**

### CODE REVIEW

File: template.tpl [hunk 1]
```diff
@@ -1624,15 +1623,14 @@ const setFromPath = (path, val, obj, target) => {
...
  *   interprets its value in (event)object instead.
  *
- * @param val {string} - the value to interpret
- * @param typ {string} - the type (one of number, boolean, string, default)
- * @param ref {string} - singifies whether val refers to a name in evObj
- * @param evObj {Object} - the object ref may refer to.
+ * @param {*} val - the value to interpret
+ * @param {string} typ - the type (one of number, boolean, string, default)
+ * @param {string} ref - singifies whether val refers to a name in evObj
+ * @param {Object} evObj - the object ref may refer to.
```
Comment (jethron) [inline template.tpl:1633]: Any idea why many of the existing comments have flipped position between name and type? From docs your change looks like the correct format, but the rest of the file is backwards for some reason. :confused: Did something change in a version or something?

File: template.tpl [hunk 2]
```diff
@@ -1679,25 +1677,28 @@ const mkStandardPairs = (evObj, evType, tagConfig) => {
...
  * Given a client event, makes a self-describing Snowplow event
  *
- * @param evObj {Object} - the client event object
- * @param evSchema {string} - the schema for the event
- * @param mappings {Array} - the event data mappings
- * @param tagConfig {Object} - the tag configuration data
- * @returns - the self-describing Snowplow event object
+ * @param {Object} evObj - the client event object
+ * @param {string} evSchema - the schema for the event
+ * @param {string[] | Object[]} mappings - the event data mappings
```
Comment (jethron) [inline template.tpl]: I think JSDoc requires parens for unions? `{(string[] | Object[])}`

**Review by jethron — APPROVED** [review]
LGTM, thanks!

I think this makes more sense that you should have to specify the type needs to cast as string if you want that behaviour, rather than assuming so and forcing it. I guess a missing feature is a `json` type that takes a string and runs it through `JSON.parse()` or something to get the real result? Not sure how useful that would be in the real world, though.

File: template.tpl (hunk 1 above)
Comment (adatzer) [reply template.tpl:1633]: > Did something change in a version or something?

Not sure. Probably just my fault.

**Comment by adatzer** [comment]
Thank you for reviewing @jethron !
The `json` type is a nice idea! Haven't implemented it for this release, but let's definitely consider it for future!

**Pull request merged**